demo-app/src/App.js

```diff
@@ -22,16 +26,61 @@ function App() {
     setSearchResults(query);
   };
 
+  const openDialog = (movie) => {
+    setSelectedMovie(movie);
+    setIsDialogOpen(true);
+  };
+
+  const closeDialog = () => {
+    setSelectedMovie(null);
+    setIsDialogOpen(false);
+  };
+
+  const handleFormSubmit = (formData) => {
+    console.log('Form data submitted:', formData);
+    closeDialog();
+  };
+
+  const handleDeleteClick = (movie) => {
+    openDialog(movie);
+  };
+
+  const handleDeleteConfirm = () => {
+    console.log('Movie deleted:', selectedMovie);
+    closeDialog();
+  };
+
   return (
     <div className="App">
+      <div>
+        <button onClick={() => openDialog()}>Add Movie</button>
+        {isDialogOpen && (
+          <ModalDialog title={selectedMovie ? 'Edit Movie' : 'Add Movie'} onClose={closeDialog}>
+            {selectedMovie ? (
+              <div>
+                <p>Are you sure you want to delete this movie?</p>
+                <button onClick={handleDeleteConfirm}>Delete</button>
+              </div>
+            ) : (
+              // Content for the movie form
+              <MovieForm initialMovieInfo={selectedMovie} onSubmit={handleFormSubmit} />
+            )}
+          </ModalDialog>
+        )}
+      </div>
+
+      <MovieForm onSubmit={handleFormSubmit} />
```

Comment:
This may be here only to check the UI.

Next, you need to open the portal when you click on the Edit button (submit the form with filled fields for editing) or Delete (question to confirm deletion).